packages/edit-site/src/components/navigation-sidebar/navigation-toggle/index.js

```diff
@@ -35,6 +35,16 @@ function NavigationToggle( { icon, isOpen } ) {
 
 	const { setIsNavigationPanelOpened } = useDispatch( editSiteStore );
 
+	document.addEventListener(
+		'keydown',
+		function ( e ) {
+			if ( e.key === 'Escape' ) {
```

Comment:
Also in that example, we can import a constant for this keyCode:

https://github.com/WordPress/gutenberg/blob/5ae2b1b06666ad1a75d5e92bfbdec4b4e728b071/packages/edit-site/src/components/secondary-sidebar/list-view-sidebar.js#L18

Reply:
@Addison-Stavlo  Hello ) Thank you for help ) I made function as you say , it works , but also this fucntion will be called even when nav panel will be closed

Reply:
>  but also this fucntion will be called even when nav panel will be closed

What do you think about adding it to the wrapper of the `NavigationPanel` component, instead of the `NavigationToggle` ?  I think that should fix that issue.

Reply:
@Addison-Stavlo if function is in NavigationPanel esc key not listening , its listening only when you click on menu , and only after that esc function will be called

Reply:
@Addison-Stavlo esc button should be listened in all document , not in part of document

Reply:
@Addison-Stavlo  can you help me with another PR ? There are no reviewers by default - https://github.com/WordPress/gutenberg/pull/29593

Reply:
>  if function is in NavigationPanel esc key not listening , its listening only when you click on menu , and only after that esc function will be called

Where it is right now, its only 'listening' if the event happens inside the toggle button.  Currently its not working as well because the keycode constant isn't imported and `ESCAPE` is still undefined.  After adding that, this approach is only closing the panel when escape is pressed inside the toggle... but if we are focused inside the panel itself, escape does nothing.

> esc button should be listened in all document , not in part of document

Id agree that global keyboard shortcuts should listen throughout the document, but escape isn't a keyboard shortcut that is specific to this panel.  Its more of an accessible way to close it for people using keyboard navigation inside the panel (as opposed to having to navigate all the way back up to the toggle button). There are other panels and dropdowns in the project that work the same way, where 'escape' is only listened for inside that specific region.  This way the context of what 'escape' is meant to close is more clear.  If we were adding a keyboard shortcut to toggle the panel, I would agree it should listen throughout the document and not be restricted to a region.

> can you help me with another PR ? There are no reviewers by default

Sure thing!  Il add some reviewers for it, and hopefully take a look early next week. 😁

Reply:
@Addison-Stavlo , I updated my PR , what do you say now ?)